### PR TITLE
chore(http): migrate from structopt to clap3

### PR DESCRIPTION
**Pull request by MarinPostma**
migrate from structopt to clap3

This fix the long lasting issue with flags require a value, such as `--no-analytics` or `--schedule-snapshot`.

All flag arguments now take NO argument, i.e:
`meilisearch --schedule-snapshot true` becomes `meilisearch --schedule-snapshot`

as per https://docs.rs/clap/latest/clap/struct.Arg.html#method.env, the env variable is defines as:
> A false literal is n, no, f, false, off or 0. An absent environment variable will also be considered as false. Anything else will considered as true.

@gmourier 
@curquiza 
@meilisearch/docs-team

### CODE REVIEW

**Comment by MarinPostma** [comment]
ping @Kerollmops just to make sure that the `no-analytics` behavior is correct

**Comment by irevoire** [comment]
:clap: :clap: :clap:

File: meilisearch-http/src/option.rs
Comment (Kerollmops) [inline meilisearch-http/src/option.rs:41]: I am at the regret to say that you must keep it like before, use an `Option<Option<bool>>` to keep the correct behavior: being able to specify `true` or `false` or nothing to enable the analitycs.

**Review by Kerollmops — CHANGES_REQUESTED** [review]
I am quite happy with this PR, thank you!

File: meilisearch-http/src/option.rs
Comment (MarinPostma) [reply meilisearch-http/src/option.rs:41]: The goal is not to have to specify true or false, i.e is no analytics is not passed, then analytics are enabled, and to disable them you need to pass the flag

Comment (gmourier) [reply meilisearch-http/src/option.rs:41]: @nicolasvienot I would like to have your opinion on this 🤓

> All flag arguments now take NO argument, i.e:
> meilisearch --schedule-snapshot true becomes meilisearch --schedule-snapshot

What do you think of this? Do you see a limit in your use of the CLI, I'm thinking mostly of the manipulation of containers dynamically within kubernetes.

Comment (Kerollmops) [reply meilisearch-http/src/option.rs:41]: Yup, by using an `Option<Option<bool>>` you do both: specify `true` or `false` and also not specify anything. It is the behavior explained in the documentation and the behavior that @nicolasvienot wanted for the SaaS. Waiting for Guigui about what we should do, keep or change.

**Comment by curquiza** [comment]
@meilisearch/docs-team, I will create a detailed issue regarding this :)

File: meilisearch-http/src/option.rs
Comment (nicolasvienot) [reply meilisearch-http/src/option.rs:41]: Hey @gmourier,

I don't think this is an issue for the manipulation of containers dynamically within Kubernetes or MeiliSearch Cloud.
We should be able to add the flag conditionally, even if this was a bit easier with `true` / `false`.

As long as this seems better for the core team and it's the same behavior everywhere, I see no objection. 😃 

@Kerollmops I made the issue about the `MEILI_NO_ANALYTICS` because I was following the doc and it seemed wrong, but we actually handle both for now because we still support < `0.25.0` versions.

Comment (MarinPostma) [reply meilisearch-http/src/option.rs:41]: @nicolasvienot, there is a misanderstanding. For the environment variable, the behaviour is the following:
> A false literal is n, no, f, false, off or 0. An absent environment variable will also be considered as false. Anything else will considered as true.
that means that a set empty env variable is considered a true as well

for the cli flag, the flag commands do not accept an argument anymore.

If you plan on only using the env variables, this should not change anything for you.

Comment (nicolasvienot) [reply meilisearch-http/src/option.rs:41]: @MarinPostma Thanks for the clarification. I just checked and as you said, since we are using env variables for now, this will still work fine. :seal:

Comment (gmourier) [reply meilisearch-http/src/option.rs:41]: Thanks, @nicolasvienot, and @MarinPostma!

@curquiza Do you think we can plan it for v0.26.0?

**Comment by MarinPostma** [comment]
@curquiza @Kerollmops, can we merge that?

**Review by Kerollmops — CHANGES_REQUESTED** [review]
I am OK to merge this, just make sure to fix Clippy.

**Review by curquiza — APPROVED** [review]
Thanks for this, sorry for the delay! I've created the related issue https://github.com/meilisearch/MeiliSearch/issues/2097

We can merge this when the clippy error is fixed!

**Review by Kerollmops — APPROVED** [review]
I approve and merge this PR, thank you!
bors merge

**Comment by MarinPostma** [comment]
bors merge

**Pull request merged**
